server/serializer/serializers.go

```diff
@@ -0,0 +1,124 @@
+package serializer
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/src-d/code-annotation/server/model"
+)
+
+// HTTPError defines an Error message as it will be written in the http.Response
+type HTTPError interface {
+	error
+	StatusCode() int
+}
+
+// Response encapsulate the content of an http.Response
+type Response struct {
+	Status int         `json:"status"`
+	Data   interface{} `json:"data,omitempty"`
+	Errors []HTTPError `json:"errors,omitempty"`
+}
+
+type httpError struct {
+	Status  int    `json:"status"`
+	Title   string `json:"title"`
+	Details string `json:"details,omitempty"`
+}
+
+// StatusCode returns the Status of the httpError
+func (e httpError) StatusCode() int {
+	return e.Status
+}
+
+// Error returns the string content of the httpError
+func (e httpError) Error() string {
+	if msg := e.Title; msg != "" {
+		return msg
+	} else if msg = http.StatusText(e.Status); msg != "" {
```

Comment:
for the future: you don't need to use `else` if you `return` before

Reply:
It was done to reuse the `msg` implicit definition made by the first `if`.
Anyhow, there is no necessity of reusing it, so I'll separate it into two different `if`, and redefine the `msg`

Reply:
done by https://github.com/src-d/code-annotation/pull/37/commits/39208c342200a46bde038d97edcc4d8b68f05975